### ISSUE TITLE
SoftmaxCrossEntropy Loss & prediction needs reworking

### ISSUE THREAD

**Issue by baykaner** [fetchai/ledger#1221]
Current implementation actually computes softmax then cross entropy in the loss function. These are combined in one loss function because the gradient computation can be simplified and more efficient if this is done (rather than prior operation being a softmax).

However, this means that when making predictions (without invoking the loss function), the user will be required to manually softmax the output of the neural net for the values to be sensible.

Some thought should be given to how to handle this at multiple levels (building graph with ops, invoking optimisers, invoking estimator)
